src/avatar-group.scss

```diff
@@ -0,0 +1,102 @@
+@import "./mixins";
+@import "./avatar";
+
+/*!
+.fd-avatar-group+((--individual|--group), (--xxs|--xs|--s|--m|--l|--xl|--xxl)
+*/
+$block: #{$fd-namespace}-avatar-group;
+
+.#{$block} {
+  $fd-avatar-group-item-spacing: (
+    "individual": (
+      "xs": ("margin-right": 0.0625rem),
+      "s": ("margin-right": 0.125rem),
+      "m": ("margin-right": 0.125rem),
+      "l": ("margin-right": 0.125rem),
+      "xl": ("margin-right": 0.25rem)
+    ),
+    "group": (
+      "xs": ("margin-right": -0.75rem),
+      "s": ("margin-right": -1.25rem),
+      "m": ("margin-right": -1.625rem),
+      "l": ("margin-right": -2rem),
+      "xl": ("margin-right": -2.75rem)
+    )
+  );
+  $fd-const-ratio: 1rem;
+  $fd-avatar-group-more-button-sizes: (
+    "xs": ("ratio": ($fd-const-ratio * 2), "font-size": 1rem),
+    "s": ("ratio": ($fd-const-ratio * 3), "font-size": 1.125rem),
+    "m": ("ratio": ($fd-const-ratio * 4), "font-size": 1.5rem),
+    "l": ("ratio": ($fd-const-ratio * 5), "font-size": 2.25rem),
+    "xl": ("ratio": ($fd-const-ratio * 7), "font-size": 3rem)
+  );
+
+  @include fd-reset();
+
+  &--individual {
+    .#{$block}__item {
+      @include fd-fiori-focus();
+      @include action-cursor();
+    }
+  }
+
+  &--group {
+    @include fd-fiori-focus();
+    @include action-cursor();
+
+    .#{$block}__more-button {
+      pointer-events: none;
+    }
+  }
+
+  &__item {
+    @include fd-reset();
+
+    display: inline-block;
+  }
+
+  &__more-button {
+    @include fd-reset();
+
+    .fd-button__text {
+      font-size: inherit;
```

Comment:
Cannot import button component inside Avatar Group component. If you need to change the properties of a button you need a modifier class like: `.fd-avatar-group__button` and then on your button element you will have `<button class="fd-button fd-avatar-group__button">...</button>`

Reply:
@InnaAtanasova that's exactly what I've done: 

```<button class="fd-button fd-avatar-group__more-button fd-avatar-group__more-button--s"></button>```

https://github.com/SAP/fundamental-styles/pull/2066/files/9a4a0051f56d36b4816bc2bc8d40c815c7f6db11#diff-c517d72c23b1a1678fc295227f77f8c2ddf4657e02f3bdbb976615056dccfe96R233

And `.fd-button-text` have this styles changes just if it placed inside `.fd-avatar-group__more-button`

Reply:
I disagree with you :) I see that you use specificity and in other applications this would work perfectly, but that's not how it works in styles. `.fd-button__text` means that an element `__text` belongs to a block `.fd-button` aka button. This means that ANY styling related to this element should only be present in the fd-button scss file. These are the rules for being self-contained. IF in another component (Avatar, Bar, Card, etc.) you use button, but you need to overwrite a little bit the text, you need an additional modifier class and you cannot rely on the specificity for this. So, in your case you will have `fd-button__text .fd-avatar-group__button-text` and inside src/avatar-group.scss file you will only have:
```
.fd-avatar-group__button-text {
     font-size: inherit;
}
```

In general, you know you are not violation the self-contained rules if inside your scss file you have only the block element `.fd-avatar-group` and inside its curly braces ALL block codes start with `__some-element` or `--some-modifier`. Hope this is clear.

Reply:
@InnaAtanasova now I got it :smiley:  Thank you!